game/dota_addons/dota_imba/scripts/vscripts/hero/hero_shadow_shaman.lua

```diff
@@ -0,0 +1,203 @@
+-- Author: Fudge
+
+CreateEmptyTalents("shadow_shaman")
+
+--------------------------------
+------ MASS SERPENT WARD -------
+--------------------------------
+imba_shadow_shaman_mass_serpent_ward = imba_shadow_shaman_mass_serpent_ward or class({})
+
+LinkLuaModifier("modifier_imba_mass_serpent_ward",  "hero/hero_shadow_shaman", LUA_MODIFIER_MOTION_NONE)
+
+-- TODO: make count as hero damage
+
+function imba_shadow_shaman_mass_serpent_ward:OnSpellStart()
+    -- Ability properties
+    local caster            =   self:GetCaster()
+    local spawn_point       =   self:GetCursorPosition()
+    local ward_level        =   self:GetLevel()
+    local ward_name         =   "npc_imba_dota_shadow_shaman_ward_"
+    local response          =   "shadowshaman_shad_ability_ward_"
+    local spawn_particle    =   "particles/units/heroes/hero_shadowshaman/shadowshaman_ward_spawn.vpcf"
+    -- Ability parameters
+    local ward_count    =   self:GetSpecialValueFor("ward_count")
+    local ward_duration =   self:GetSpecialValueFor("duration")
+
+    -- Emit cast sound
+    caster:EmitSound("Hero_ShadowShaman.SerpentWard")
+    -- Emit cast response
+    caster:EmitSound(response..RandomInt(4,7))
+
+    -- Emit spawn particle
+    local spawn_particle_fx = ParticleManager:CreateParticle(spawn_particle, PATTACH_ABSORIGIN, caster)
+    ParticleManager:SetParticleControl( spawn_particle_fx, 0, spawn_point )
+
+    -- Create 8 wards in a box formation (note the vectors)
+    ward_table  =   {}
+    ward_table[1] = CreateUnitByName(ward_name..ward_level, spawn_point + Vector(-64, -64 ,0), true, caster, caster, caster:GetTeamNumber())
```

Comment:
Can't this be done in a loop?
Also, can we make this clearer?

```lua
vecs = {
    Vector(-64, -64, 0),
    Vector(-64, 0, 0),
    Vector(-64, 64, 0),
    Vector(0, -64 ,0),
    Vector(0, 64 ,0),
    Vector(64, -64 ,0),
    Vector(64, 0 ,0),
    Vector(64, 64 ,0),
    Vector(-128 , 64 ,0),
    Vector(128 , 64 ,0)
}

for k, vec in pairs(vecs) do
    local unit_name = ward_name..ward_level
    local location = spawn_point + vec
    local find_clear_space = true
    local npc_owner = caster
    local unit_owner = caster
    local team_number = caster:GetTeamNumber()
    ward_table[k] = CreateUnitByName(unit_name, location, find_clear_space, npc_owner, unit_owner, team_number)
end
```

This example gets rid of repeated code, so if you ever need to change something you just need to change it in one place. You can also easily inject/add new Vectors if needed.